db/migrate/20170519153350_add_showback_models.rb

```diff
@@ -0,0 +1,73 @@
+class AddShowbackModels < ActiveRecord::Migration[5.0]
+  def up
+    create_table :showback_usage_types, :id => :bigserial, :force => :cascade do |t|
+      t.string     :category, :allow_nil => false
+      t.string     :description
+      t.string     :measure, :allow_nil => false
+      t.text       :dimensions, :array => true, :allow_nil => false, :default => []
+      t.timestamp  :updated_at
+      t.timestamp  :created_at
+    end
+
+    create_table :showback_events do |t|
+      t.json       :data,       :allow_nil => false, :default => {}
+      t.timestamp  :start_time, :allow_nil => false  # When the event start
+      t.timestamp  :end_time,   :allow_nil => false  # When the event finish
+      t.belongs_to :resource,   :allow_nil => false, :type => :bigint, :polymorphic => true
+      t.json       :context,    :allow_nil => false, :default => {}
+      t.timestamp  :updated_at
+      t.timestamp  :created_at
+    end
+    add_index  :showback_events, :resource_id
+    add_index  :showback_events, :resource_type
+
+    create_table :showback_price_plans, :id => :bigserial, :force => :cascade do |t|
+      t.string     :name, :allow_nil => false
+      t.belongs_to :resource, :allow_nil => false, :type => :bigint, :polymorphic => true
+      t.string     :description
+      t.timestamps
+    end
+
+    create_table :showback_rates, :id => :bigserial, :force => :cascade do |t|
+      t.monetize   :fixed_rate
+      t.monetize   :variable_rate
+      t.string     :calculation,   :allow_nil  => false
+      t.string     :category,      :allow_nil  => false
+      t.string     :dimension,     :allow_nil  => false
+      t.jsonb      :screener,      :allow_nil  => false, :default => {}
+      t.datetime   :date
+      t.string     :concept
+      t.belongs_to :showback_price_plan, :type => :bigint
+      t.timestamps
+    end
+    add_index :showback_rates, :screener, using: :gin
+    add_index :showback_rates, [:category, :dimension]
+
+    create_table :showback_buckets, :id => :bigserial, :force => :cascade do |t|
+      t.string     :name, :allow_nil  => false
+      t.string     :description
+      t.timestamp  :start_time,   :allow_nil  => false
+      t.timestamp  :end_time,     :allow_nil  => false
+      t.string     :state,        :allow_nil  => false
+      t.monetize   :accumulated_cost
+      t.references :resource, :polymorphic => true, :type => :bigint, :index => true
+      t.timestamps
+    end
+
+    create_table :showback_charges, :id => :bigserial, :force => :cascade do |t|
+      t.monetize   :cost
```

Comment:
What does monetize do?  Can we get away with just creating the columns manually instead of needing an extension library?